### DISCUSSION TITLE
Suspicious code in telegram/updates/manager.go

### DISCUSSION THREAD

**Discussion by MeanSquaredError** [gotd/td#674] [Q&A] [answered]
In telegram/updates/manager.go in `func (m *Manager) loadState(...)` there is the following code
```
		state = state.fromRemote(remote)
		if err := m.cfg.Storage.SetState(userID, state); err != nil {
			return State{}, err
		}

		if err := m.cfg.Storage.SetState(userID, state); err != nil {
			return State{}, err
		}
```

Why does `m.cfg.Storage.SetState()` get called twice? Is it just a copy&paste error or is there a reason for the double call?

**Comment by ernado** [comment, marked as answer]
Looks like copy-paste

cc @tdakkota @shadowspore
